src/udev/udevadm-control.c

```diff
@@ -136,11 +137,12 @@ static int adm_control(struct udev *udev, int argc, char *argv[]) {
                 case 't': {
                         int seconds;
 
-                        seconds = atoi(optarg);
-                        if (seconds >= 0)
-                                timeout = seconds;
-                        else
+                        rc = safe_atoi(optarg, &seconds);
```

Comment:
the "return rc" should be "return r".

And I'd probably round up in the division... i.e.`(s+USEC_PER_SEC-1)/USEC_PER_SEC`, but I figure that doesn't matter much...

looks great otherwise...